### ISSUE TITLE
Store response bodies

### ISSUE THREAD

**Issue by yoavweiss** [HTTPArchive/legacy.httparchive.org#6]
It'd be great if we could save the response bodies alongside the meta data, so that they'd be available for download, and possibly for querying later on. That would obsolete slightly overlapping efforts like [webdevdata.org](http://webdevdata.org/), and would enable that group's members (e.g. me :) ) to help out with the HTTPArchive.

We should probably keep the actual response bodies off the database, and just save them to disk, while only keeping the file name in the database.

Since we're talking about a lot of data, it probably makes sense to start out with text based resources only, and see about images later on. 

Thoughts?

**Comment by h3nce** [comment]
Hi @pmeenan, 

.zip files are working up to February 2015 (http://httparchive.webpagetest.org/habodies.php?run=20150215). Any chance of a refresh for Nov-Dec 2015?

Thanks,

Tom

**Comment by pmeenan** [comment]
We are storing them in a public bucket in Google cloud storage (and will also be making it available in big query). You'll need a developer account to rsync the folders but it is free. I'll have some UI with the bucket paths shortly. The 11/15 crawl was the first one and all 4 crawls uploaded correctly and every crawl going forward should also. 






On Thu, Dec 3, 2015 at 2:22 PM -0800, "h3nce" <notifications@github.com> wrote:












Hi @pmeenan, 



.zip files are working up to February 2015 (http://httparchive.webpagetest.org/habodies.php?run=20150215). Any chance of a refresh for Nov-Dec 2015?



Thanks,



Tom

**Comment by h3nce** [comment]
Super, thank you for the update. Yes, I was using the BigQuery UI, that one has response bodies up to Aug 2014 currently. I'll postpone my .zip d/l which is still going from time of posting :)

**Comment by pmeenan** [comment]
Still working on the real documentation and the UI for showing all of the
directory names but if you want to grab the latest crawl, you'll need to
install gsutil <https://cloud.google.com/storage/docs/gsutil_install>, set
up credentials and then run:

gsutil -m rsync gs://httparchive/desktop-Nov_15_2015 .

Each HAR is a separate gzip file (~490k of them). The -m is important as it
will run multiple downloads in parallel, otherwise it will take eons.  Hope
you gave a good internet connection (or are really patient) though because
it's probably > 100GB.

Let us know if there are issues with credentials.  The buckets are all
public so you should be able to sync (and individual files are available
over plain HTTPS) but I haven't tried with an account that doesn't already
have write permissions to the bucket.

Thanks,

-Pat

**Comment by foolip** [comment]
What is the difference between gs://httparchive/Dec_1_2015/, gs://httparchive/desktop-Dec_1_2015/ and gs://httparchive/mobile-Dec_1_2015/ ?

Is the first just the union of the desktop and mobile sets?

**Comment by foolip** [comment]
> I haven't tried with an account that doesn't already have write permissions to the bucket.

FWIW, I can confirm that I was able to sync the whole desktop-Dec_1_2015 dataset.

**Comment by RByers** [comment]
Looks like the simple folders like 'Jan_1_2017' just have summary info (pages.csv.gz).  And instead of 'desktop' and 'mobile' the options are now 'chrome' and 'android'.  Looks like the latest desktop data is in `gs://httparchive/chrome-Jan_1_2017`.
